packages/dht/src/transport/ListeningRpcCommunicator.ts

```diff
@@ -4,11 +4,19 @@ import { RpcCommunicatorConfig } from '@streamr/proto-rpc'
 import { Message } from '../proto/packages/dht/protos/DhtRpc'
 
 export class ListeningRpcCommunicator extends RoutingRpcCommunicator {
+    private readonly transport: ITransport | undefined
+    private readonly handler: (msg: Message) => void
     constructor(ownServiceId: string, transport: ITransport, config?: RpcCommunicatorConfig) {
         super(ownServiceId, transport.send, config)
-        
-        transport.on('message', (msg: Message) => {
+        this.handler = (msg: Message) => {
             this.handleMessageFromPeer(msg)
-        })
+        }
+        this.transport = transport
+        transport.on('message', this.handler) 
+    }
+
+    destroy(): void {
+        this.transport!.off('message', this.handler)
+        this.stop()
```

Comment:
Could rename `stop` to `destroy` in `RpcCommunicator` (maybe in a separate PR)

Reply:
Yeah, at the moment the ListeningRcCommunicator calls stop from the super class in destroy. Would probably make sense to get rid of inheritance here